spec/models/advance_on_docket_motion_spec.rb

```diff
@@ -120,58 +120,133 @@
-        let(:reason) {  described_class.reasons[:age] }
-
+      context "when the motion is age-related" do
+        let(:reason) { described_class.reasons[:age] }
         it "creates a new motion" do
           subject
           motions = appeal.claimant.person.advance_on_docket_motions
-          expect(motions.count).to eq 2
-          expect(motions.first.granted).to be(false)
+          expect(motions.count).to eq 1
+          expect(motions.first.granted).to be(true)
           expect(motions.first.reason).to eq(described_class.reasons[:age])
-          expect(motions.second.granted).to be(true)
-          expect(motions.second.reason).to eq(attrs[:reason])
         end
       end
     end
+
+    context "when there is an existing motion for the appeal" do
+      let(:appeal_on_motion) { appeal }
+
+      context "when the previous motion reason is not age" do
+        # Because we're mostly testing updates, create an initial AOD motion first:
+        before do
+          described_class.create!(
+            person_id: claimant.person.id,
+            granted: false,
+            reason: described_class.reasons[:other],
+            user_id: user_id,
+            appeal: appeal_on_motion
+          )
+        end
+
+        context "when called with an age-related motion" do
+          let(:reason) { described_class.reasons[:age] }
+
+          it "creates a new age-related motion" do
+            subject
+            motions = appeal.claimant.person.advance_on_docket_motions
+            expect(motions.count).to eq 2
+            expect(motions.first.granted).to be(false)
+            expect(motions.first.reason).to eq(described_class.reasons[:other])
+            expect(motions.second.granted).to eq(true)
+            expect(motions.second.reason).to eq(described_class.reasons[:age])
+          end
+        end
+
+        context "when called with a non-age-related motion" do
+          let(:reason) { described_class.reasons[:serious_illness] }
+
+          it "updates the previous motion" do
+            subject
+            motions = appeal.claimant.person.advance_on_docket_motions
+            expect(motions.count).to eq 1
+            expect(motions.first.granted).to be(true)
+            expect(motions.first.reason).to eq(described_class.reasons[:serious_illness])
+          end
+        end
+      end
+
+      context "when the previous motion reason is age" do
+        # Because we're mostly testing updates, create an initial AOD motion first:
+        before do
+          described_class.create!(
+            person_id: claimant.person.id,
+            granted: false,
+            reason: described_class.reasons[:age],
+            user_id: user_id,
+            appeal: appeal_on_motion
+          )
+        end
+
+        context "when called with an age-related motion" do
+          let(:reason) { described_class.reasons[:age] }
+
+          it "only allows one age-related motion per appeal" do
```

Comment:
Match the wording above?
```suggestion
          it "updates the existing motion" do
```